File: georocket-server-api/src/main/java/io/georocket/index/IndexerFactory.java
package io.georocket.index;

import io.georocket.query.QueryCompiler;

/**
 * Factory for {@link Indexer} objects
 * @since 1.0.0
 * @author Michel Kraemer
 */
public interface IndexerFactory extends QueryCompiler {
  /**
   * @return a new instance of {@link Indexer}
   */
  Indexer createIndexer();
}

File: georocket-server/src/main/kotlin/io/georocket/index/IndexerFactory.kt
package io.georocket.index

import io.georocket.query.QueryCompiler
import io.georocket.util.StreamEvent

/**
 * Factory for [Indexer] objects
 * @author Michel Kraemer
 */
interface IndexerFactory : QueryCompiler {
  /**
   * Returns a new instance of [Indexer] for the given [eventType]. Returns
   * `null` if the type is not supported.
   */
  fun <T : StreamEvent> createIndexer(eventType: Class<T>): Indexer<T>?
}
